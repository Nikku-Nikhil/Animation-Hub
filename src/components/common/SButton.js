import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import React from 'react';
import SText from './SText';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

const SButton = ({
  text,
  textSize = 14,
  backgroundColor = '#eee',
  color = '#000',
  width = '100%',
  containerStyle,
  iconShown = true,
  borderColor = '#000',
  icon = <MaterialIcons name="arrow-forward-ios" size={20} />,
  onPress = () => {},
}) => {
  console.log("Text",text);
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[
        {
          backgroundColor,
          width,
          borderColor,
        },
        styles.btn,
        containerStyle,
      ]}>
      <SText text={text} color={color} size={textSize} />
      {iconShown && icon}
    </TouchableOpacity>
  );
};

export default SButton;

const styles = StyleSheet.create({
  btn: {
    padding: 18,
    justifyContent: 'space-between',
    flexDirection: 'row',
    borderRadius: 8,
    borderWidth: 1,
    alignItems: 'center',
    marginVertical: 5,
  },
});
