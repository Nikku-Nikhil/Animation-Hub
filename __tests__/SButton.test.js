import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import SButton from '../src/components/common/SButton';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'MaterialIcons');

const initialProps={
    text:'Nikhil',
    textSize : 14,
    backgroundColor : '#eee',
    color : '#000',
    width : '100%',
    containerStyle:{},
    iconShown : true,
    borderColor : '#000',
    icon : <MaterialIcons name="arrow-forward-ios" size={20} />,
}

describe('SButton component', () => {

  test("First taking snapshot",()=>{
    const snapshot = render(<SButton {...initialProps} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  })  


  it.only('Check if the SButton renders with correct text', () => {
    const expectedText = 'hello';
    const screen = render(<SButton text={expectedText} />);    
    const button = screen.getByText(expectedText);
    expect(button).toBeTruthy();
  });
  

  test('renders correctly with default props', () => {
    const { getByText } = render(<SButton text="Click me" />);
    const buttonText = getByText('Click me');
    expect(buttonText).toBeTruthy();
  });

  test('fires onPress callback when pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<SButton text="Click me" onPress={onPressMock} />);
    const button = getByText('Click me');
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  test('renders icon when iconShown is true', () => {
    const { getByTestId } = render(<SButton text="Click me" iconShown={true} />);
    expect(getByTestId).toBeTruthy();
  });

  test('does not render icon when iconShown is false', () => {
    const { queryByTestId } = render(<SButton text="Click me" iconShown={false} />);
    const icon = queryByTestId("");
    expect(icon).toBeNull();
  });

  test('renders with custom backgroundColor, color, width, and borderColor props', () => {
    const { toJSON } = render(
      <SButton
        text="Custom Style"
        backgroundColor="#FF5733"
        color="#FFFFFF"
        width="50%"
        borderColor="#0088CC"
      />
    );
    const componentTree = toJSON();
    expect(componentTree).toMatchSnapshot(); 
  });

 
});
