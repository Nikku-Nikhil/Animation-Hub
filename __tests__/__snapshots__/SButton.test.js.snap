// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SButton component First taking snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderColor": "#000",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 5,
      "opacity": 1,
      "padding": 18,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": 500,
          "lineHeight": undefined,
        },
        undefined,
      ]
    }
  />
  <MaterialIcons
    name="arrow-forward-ios"
    size={20}
  />
</View>
`;

exports[`SButton component renders with custom backgroundColor, color, width, and borderColor props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF5733",
      "borderColor": "#0088CC",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 5,
      "opacity": 1,
      "padding": 18,
      "width": "50%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": 500,
          "lineHeight": undefined,
        },
        undefined,
      ]
    }
  >
    Custom Style
  </Text>
  <MaterialIcons
    name="arrow-forward-ios"
    size={20}
  />
</View>
`;
